### PR TITLE
[ASCII-1992] Use the new tunneled http client for sending requests from e2e hosts

### PR DESCRIPTION
**Pull request by jeremy-hanna**
### What does this PR do?

Use the new tunneled http client for sending requests from e2e hosts instead of curl executions

### CODE REVIEW

File: test/new-e2e/tests/agent-shared-components/api/api_test.go [hunk 1]
```diff
@@ -406,15 +404,22 @@ func (v *apiSuite) TestDefaultAgentAPIEndpoints() {
...
+			assert.NoError(t, err, "failed to create request")
+
+			hostHTTPClient := host.NewHTTPClient()
 			require.EventuallyWithT(t, func(ct *assert.CollectT) {
-				resp, err := host.Execute(cmd)
-				if assert.NoError(ct, err) {
-					testcase.assert(ct, testcase.agentEndpointInfo, resp)
-				}
+				resp, err := hostHTTPClient.Do(req)
+				assert.NoError(ct, err, "failed to send request")
```
Comment (pgimalac) [inline test/new-e2e/tests/agent-shared-components/api/api_test.go:416]: **❓ question:** ‏Should we check the http status code ? (it depends on the tests I guess)
`Do` only returns an error if there's no response / the response is invalid, but it doesn't complain when the response is a 4xx or 5xx

Comment (pgimalac) [reply test/new-e2e/tests/agent-shared-components/api/api_test.go:416]: Same comment for other instances below

**Review by pgimalac — APPROVED** [review]
LGTM, just a few open questions/nitpicks

File: test/new-e2e/tests/agent-shared-components/api/api_test.go [hunk 2]
```diff
@@ -55,24 +58,19 @@ func (endpointInfo *agentEndpointInfo) url() *url.URL {
 	}
 }
 
-func (endpointInfo *agentEndpointInfo) fetchCommand(authtoken string) string {
+func (endpointInfo *agentEndpointInfo) httpRequest(authtoken string) (*http.Request, error) {
 	data := endpointInfo.data
 	if len(endpointInfo.data) == 0 {
 		data = "{}"
 	}
```
[the remark this answers is not in the document]
Comment (jeremy-hanna) [reply test/new-e2e/tests/agent-shared-components/api/api_test.go:65]: the internal api will blindly try to unmarshal json ([example](https://github.com/DataDog/datadog-agent/blob/main/comp/core/flare/flare.go#L123)) when sent a request in unexpected places, this is largely undocumented so we have to provide an empty json object in a few places

I think just sending an empty io.Reader should work and we can remove this 🤔

File: test/new-e2e/tests/agent-shared-components/api/api_test.go (hunk 1 above)
Comment (jeremy-hanna) [reply test/new-e2e/tests/agent-shared-components/api/api_test.go:515]: fixed in https://github.com/DataDog/datadog-agent/pull/27685/commits/0e92ac6f1f91831ff126803482cb50df8e64962e

File: test/new-e2e/tests/agent-shared-components/api/api_test.go [hunk 3]
```diff
@@ -406,15 +404,22 @@ func (v *apiSuite) TestDefaultAgentAPIEndpoints() {
...
 
+	host := v.Env().RemoteHost
 	for _, testcase := range testcases {
-		cmd := testcase.fetchCommand(authtoken)
-		host := v.Env().RemoteHost
 		v.T().Run(fmt.Sprintf("API - %s test", testcase.name), func(t *testing.T) {
+			req, err := testcase.httpRequest(authtoken)
+			assert.NoError(t, err, "failed to create request")
+
+			hostHTTPClient := host.NewHTTPClient()
```
[the remark this answers is not in the document]
Comment (jeremy-hanna) [reply test/new-e2e/tests/agent-shared-components/api/api_test.go]: fixed in https://github.com/DataDog/datadog-agent/pull/27685/commits/0e92ac6f1f91831ff126803482cb50df8e64962e

File: test/new-e2e/tests/agent-shared-components/api/api_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeremy-hanna) [reply test/new-e2e/tests/agent-shared-components/api/api_test.go]: fixed in https://github.com/DataDog/datadog-agent/pull/27685/commits/0e92ac6f1f91831ff126803482cb50df8e64962e

**Pull request merged**